src/lockdown.cpp

```diff
@@ -0,0 +1,87 @@
+#include <algorithm>
+#include <fstream>
+
+#include <cerrno>
+
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/utsname.h>
+#include <unistd.h>
+
+#include "lockdown.h"
+
+namespace bpftrace {
+namespace lockdown {
+
+static LockdownState from_string(const std::string &s)
+{
+  if (s == "none")
+    return LockdownState::None;
+  else if (s == "integrity")
+    return LockdownState::Integrity;
+  else if (s == "confidentiality")
+    return LockdownState::Confidentiality;
+
+  return LockdownState::Unknown;
+}
+
+static bool is_ubuntu(void)
+{
+  // If ubuntu is somewhere in uname it is probably ubuntu
+  struct utsname name = {};
+  uname(&name);
+
+  std::string version(name.version);
+
+  std::transform(version.begin(),
+                 version.end(),
+                 version.begin(),
+                 [](unsigned char c) { return std::tolower(c); });
+
+  return (version.find("ubuntu") != std::string::npos);
+}
+
+static LockdownState read_security_lockdown(void)
+{
+  std::ifstream file("/sys/kernel/security/lockdown");
+  if (file.fail())
+    return LockdownState::Unknown;
+
+  // Format: none [integrity] confidentiality
+  while (!file.fail())
+  {
+    std::string line;
```

Comment:
This loop was a little confusing since the temp var was named `line` (also the entire `std::*stream` API is confusing). Maybe use `split_string()` or name this `entry`.

Reply:
ill rename it to field an add an extra comment.

Not a fan of `*stream` either but I think this works ok in this case.